airflow/jobs/scheduler_job.py

```diff
@@ -721,25 +721,14 @@ def _process_task_instances(self, dag, task_instances_list, session=None):
             if run.state == State.RUNNING:
                 make_transient(run)
                 active_dag_runs.append(run)
-
-        for run in active_dag_runs:
-            self.log.debug("Examining active DAG run: %s", run)
-            tis = run.get_task_instances(state=SCHEDULEABLE_STATES)
-
-            # this loop is quite slow as it uses are_dependencies_met for
-            # every task (in ti.is_runnable). This is also called in
-            # update_state above which has already checked these tasks
-            for ti in tis:
-                task = dag.get_task(ti.task_id)
-
-                # fixme: ti.task is transient but needs to be set
-                ti.task = task
-
-                if ti.are_dependencies_met(
-                        dep_context=DepContext(flag_upstream_failed=True),
-                        session=session):
+                self.log.debug("Examining active DAG run: %s", run)
+                for ti in run.ready_tis:
```

Comment:
Rather than a property on the run, lets make it the return value from `update_state` (that way we also don't have to "clear" it afterwards)

Reply:
I did It, the problem is there are plenty of tests that uses the output of dr.update_state() so I will have to change them to take state from the object

Reply:
I thought `dr.update_state` currently didn't return anything? Did I read the code wrong?

Reply:
yes :)
However just few tests use it for the state and it is kind of an easy change